api/src/pcapi/routes/public/individual_offers/v1/serialization.py

```diff
@@ -943,3 +947,24 @@ class Config:
 
 class GetOfferersVenuesQuery(serialization.ConfiguredBaseModel):
     siren: str | None = pydantic.Field(example="123456789")
+
+
+class GetProductsListByEansQuery(serialization.ConfiguredBaseModel):
+    eans: str | None = pydantic.Field(example="0123456789123,0123456789124")
+
+    @pydantic.validator("eans")
+    def ean_list_must_not_be_empty_and_ean_must_have_13_caracters(cls, eans: str) -> list:
+        ean_list = eans.split(",")
+        if len(ean_list) > 100:
+            raise ValueError("Too many EANs")
+        if len(ean_list) == 0:
+            raise ValueError("EAN list must not be empty")
+        for ean in ean_list:
+            if len(ean) != 13:
+                logger.error(ean)
+                raise ValueError("Only 13 characters EAN are accepted")
+            if not str(ean).isdigit():
+                raise ValueError("EAN must be an integer")
+            if int(ean) < 0:
+                raise ValueError("EAN must be positive")
+        return ean_list
```

Comment:
<!-- Issues represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 issue:** ‏Ici le premier `if` va toujours empêcher les 2 suivants.

```suggestion
            if not str(ean).isdigit():
                raise ValueError("EAN must be an integer")
            if int(ean) < 0:
                raise ValueError("EAN must be positive") 
            if len(ean) != 13:
                raise ValueError("Only 13 characters EAN are accepted")
        return ean_list